lib/rust/prelude/src/reference.rs

```diff
@@ -36,6 +36,24 @@ pub fn with<T, F: FnOnce(T) -> Out, Out>(t: T, f: F) -> Out {
 
 
 
+// ================
+// === TryAsRef ===
+// ================
+
+/// Just like [`AsRef`], but might return [`None`] for some data.
+#[allow(missing_docs)]
+pub trait TryAsRef<T> {
+    fn try_as_ref(&self) -> Option<&T>;
```

Comment:
`AsRef` could also be replaced with `From` and `Into`. It exists because it provides a very common signature, so the usage-places don't need to have explicit type signatures. This situation is similar. I understand that `try_` prefix is connected with `Result` more than `Option` so I'm thinking about better naming here, but I don't have a good idea yet. I'd love to chat about it.